corehq/apps/app_execution/db_accessors.py

```diff
@@ -1,26 +1,43 @@
 from django.db.models import Avg, DateTimeField, DurationField, ExpressionWrapper, F, Max
 from django.db.models.functions import Trunc
 
-from corehq.apps.app_execution.models import AppExecutionLog
+from corehq.apps.app_execution.models import AppExecutionLog, AppWorkflowConfig
 
 
-def get_avg_duration_data(workflow_id, start, end):
+def get_avg_duration_data(start, end, workflow_id=None):
+    query = AppExecutionLog.objects.filter(started__gte=start, started__lt=end)
+    if workflow_id:
+        query = query.filter(workflow_id=workflow_id)
+
     chart_logs = (
-        AppExecutionLog.objects.filter(workflow_id=workflow_id)
-        .filter(started__gte=start, started__lt=end)
-        .annotate(
+        query.annotate(
             date=Trunc("started", "hour", output_field=DateTimeField()),
             duration=ExpressionWrapper(F("completed") - F("started"), output_field=DurationField()),
-        ).values("date")
+        ).values("date", "workflow_id")
         .annotate(avg_duration=Avg('duration'))
         .annotate(max_duration=Max('duration'))
-        .order_by("date")
+        .order_by("workflow_id", "date")
     )
-    return [
-        {
+
+    data = []
+    seen_workflows = set()
+    for row in chart_logs:
+        if row["workflow_id"] not in seen_workflows:
+            seen_workflows.add(row["workflow_id"])
+            data.append({
+                "key": row["workflow_id"],
+                "values": []
+            })
+        data[-1]["values"].append({
```

Comment:
nit: you might be able to simplify this a bit by storing a `data` as a `defaultdict(list)` keyed by workflow ID, and using `keys()` and `values()` in the next bits.

Reply:
I did consider that but since I want to initialize the element with more e.g. the key, this approach worked out better.